walkstruct.go

```diff
@@ -37,6 +37,34 @@ func doWalkStructElements(t reflect.Type, path []int, f func(reflect.StructField
 	}
 }
 
+func WalkStructElementsWithError(t reflect.Type, f func(reflect.StructField) (bool, error)) error {
```

Comment:
Needs doc comment.

Suggestion change the func to `func(reflect.StructField) error` and have a new function:

```
func DoNotRecurse(error) error
```

That can be used to control recursion. `DoNotRecurse` needs to be okay with wrapping `nil` which is not considered an error.